src/views/Proposal.js

```diff
@@ -0,0 +1,319 @@
+
+import PropTypes from 'prop-types';
+import Box from '3box';
+
+import useWeb3Connect from '../utils/useWeb3Connect';
+import useRouter from '../utils/useRouter';
+import { linkTwitterHandleToEthAddressInFirebase } from '../utils/twitterUtils';
+
+import DepositIcon from '@material-ui/icons/AllInclusive';
+import TwitterIcon from '@material-ui/icons/Twitter';
+
+import { makeStyles } from '@material-ui/styles';
+import { Typography, Button, Grid } from '@material-ui/core';
+import HowToVoteIcon from '@material-ui/icons/HowToVote';
+import { IconButton, Tooltip, Link } from '@material-ui/core';
+
+import Page from '../components/Page';
+import Header from '../components/Header';
+import ProposalCard from '../components/ProposalCard';
+import EllipsisLoader from '../components/EllipsisLoader';
+
+import { twitterHandleAlreadyLinked } from '../modules/twitterDb';
+import { getUrlByHash } from '../modules/pinata';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+    position: 'relative',
+    width: '100%',
+  },
+  button: {
+    width: 190,
+  },
+  image: {
+    display: 'block',
+    maxHeight: '300px',
+    maxWidth: '100%',
+  },
+}));
+
+const Proposal = ({ match }) => {
+  const web3Connect = useWeb3Connect();
+  // const proposals = useSelector((state) => state.proposals.proposals);
+  const { proposals, fetched } = web3Connect;
+  const proposal_id = match.params.proposal_id;
+  const classes = useStyles();
+  const router = useRouter();
+  const [status, setStatus] = useState('DRAFT');
+
+  const [proposal, setProposal] = useState(null);
+
+  useEffect(() => {
+    if (fetched) {
+      setProposal(
+        proposals.find((proposal) => proposal.id == match.params.proposal_id)
+      );
+    }
+  }, [fetched]);
+
+  const canVoteWithDelegate =
+    status === 'ENABLED' ||
+    (status !== '3BOX_VERIFIED' && web3Connect.enabledTwitter);
+
+  const [canVoteViaTwitter, setCanVoteViaTwitter] = useState(false);
+
+  const address = web3Connect.address;
+
+  const enableTwitter = async () => {
+    setStatus('3BOX_VERIFICATION');
+    const profile = await Box.getProfile(address);
+    const verified = await Box.getVerifiedAccounts(profile);
+    if (verified && verified.twitter && verified.twitter.username) {
+      setStatus('3BOX_VERIFIED');
+      let tx = await web3Connect.contracts.dao.methods.enableTwitterVoting();
+      if (!tx) {
+        setStatus('TX_FAILED');
+      } else {
+        let txHash = tx.transactionHash;
+        try {
+          let result = await linkTwitterHandleToEthAddressInFirebase(
+            verified.twitter.username,
+            address,
+            txHash
+          );
+          setStatus('ENABLED');
+        } catch (error) {
+          console.error(error);
+          setStatus('TX_FAILED');
+        }
+      }
+    } else {
+      setStatus('3BOX_FAILED');
+    }
+  };
+
+  const voteTwitter = (emoji) => {
```

Comment:
This method is duplicated with the one in the ProposalCard component. It can be like that for now but in the future, we should have just one method to handle that in case we decided to change the tweet text.

Reply:
Agreed, I pushed a bit early, a few functions can move to the new twitter utils file I created